apps/passport-server/migrations/57_password_reset_limit.sql

```diff
@@ -0,0 +1,94 @@
+CREATE TABLE rate_limit_types(
+  event_type VARCHAR NOT NULL,
+  hourly_limit INT NOT NULL,
+  PRIMARY KEY(event_type)
+);
+
+CREATE TABLE rate_limit_buckets(
+  event_type VARCHAR NOT NULL,
+  event_id VARCHAR NOT NULL,
+  remaining INT NOT NULL,
+  last_refill TIMESTAMPTZ,
+  PRIMARY KEY(event_type, event_id),
+  FOREIGN KEY(event_type) REFERENCES rate_limit_types
+);
+
+-- Allow 10 password reset attempts per hour
+INSERT INTO rate_limit_types VALUES('CHECK_EMAIL_TOKEN', 10);
+INSERT INTO rate_limit_types VALUES('REQUEST_EMAIL_TOKEN', 10);
+
+-- MIT License
```

Comment:
Can you link to the repo this came from, in addition to the license?

Reply:
Done!